iothub/service/src/Configurations/ConfigurationsClient.cs

```diff
@@ -66,7 +66,7 @@ public virtual async Task<Configuration> CreateAsync(Configuration configuration
 
             try
             {
-                Argument.RequireNotNull(configuration, nameof(configuration));
+                Argument.AssertNotNull(configuration, nameof(configuration));
```

Comment:
I think we'll find it confusing when the Arguments classes per NuGet package don't work the same. I've modeled the names after what is in Track2, again for consistency.